third_party/blink/renderer/bindings/core/v8/record_replay_interface.cc

```diff
@@ -235,11 +235,33 @@ function sendMessage(method, params) {
     }
   }
   gCurrentMessageId = undefined;
-  if (gCurrentMessageResult?.result) {
-    return gCurrentMessageResult.result;
+  
+  /**
+   * `gCurrentMessageResult` can be of 3 possible types:
+   * 
+   * 1. ProtocolError (id?, error: (code, message), data?)
+   * @see https://github.com/replayio/chromium-v8/blob/c5e451943a6d87b44374e7a08d44fa92b9a2c93b/third_party/inspector_protocol/crdtp/dispatch.cc#L275
+   * 
+   * 2. Response (id, result) - The response contains the return values defined by CDP.
+   * @see https://github.com/replayio/chromium-v8/blob/c5e451943a6d87b44374e7a08d44fa92b9a2c93b/third_party/inspector_protocol/crdtp/dispatch.cc#L348
+   * 
+   * 3. Notification (method, params) - TODO: we are not handling this yet.
+   * @see https://github.com/replayio/chromium-v8/blob/c5e451943a6d87b44374e7a08d44fa92b9a2c93b/third_party/inspector_protocol/crdtp/dispatch.cc#L370
+   */
+  if (throwOnApiError) {
+    if (gCurrentMessageResult?.result) {
+      return gCurrentMessageResult.result;
+    }
+    if (gCurrentMessageResult?.error) {
+      throw new Error(`${gCurrentMessageResult.error.message} (${gCurrentMessageResult.error.code})`);
+    }
   }
-  if (gCurrentMessageResult?.error) {
-    throw new Error(`${gCurrentMessageResult.error.message} (${gCurrentMessageResult.error.code})`);
```

Comment:
Having a param like `throwOnApiError` is very unusual JS API-design-wise and I think we need to put more care into this, because we're pretty much stuck with any change once it is released.

Instead of this, I think we should instead do
```
class CDPMessageError extends Error {
  constructor(message, code) {
    super(message);
    this.code = code;
  }
}
```
outside the function and
```
throw new CDPMessageError(gCurrentMessageResult.error.message, gCurrentMessageResult.error.code);
```
here. 

Then the various callsites are free catch and to handle the error however they please with a well-defined error type, rather than an overloaded `return` type object.